File: Debug/Python/6.py
# Using Generator Function find the fibonacci series

def fibonacci_generator():
    a, b = 0, 1
    while True:
        yield a
        a, b = b, a + b

n = 10
fibonacci_gen = fibonacci_generator()
fibonacci_sequence_gen = [fibonacci_gen for _ in range(n)]
print("Fibonacci Sequence using generator function:", fibonacci_sequence_gen)

File: Debug/C++/6.cpp
// Using Generator Function find the fibonacci series

#include <iostream>

class Main6
{
private:
    int n;

public:
    Main6(int n) : n(n) {}

    class Iterator
    {
    private:
        int count;
        int a;
        int b;

    public:
        Iterator() : count(0), a(0), b(1) {}

        bool hasNext()
        {
            return count < n;
        }

        int next()
        {
            int result = a;
            a = b;
            b = result;
            return result;
        }
    };

    Iterator iterator()
    {
        return Iterator();
    }
};

int main()
{
    int n = 10;
    Main6 generator(n);
    Main6::Iterator it = generator.iterator();
    while (it.hasNext())
    {
        std::cout << it.next() << " ";
    }
    return 0;
}
